Code/autopkglib/recipes/__init__.py

```diff
@@ -0,0 +1,762 @@
+    return None
+
+
+def find_recipe_by_id_in_map(
+    identifier: str, skip_overrides: bool = False
+) -> Optional[str]:
+    """Search recipe map for an identifier"""
+    if not skip_overrides and identifier in globalRecipeMap.get(
+        "overrides-identifiers", {}
+    ):
+        log(f"Found {identifier} in recipe map overrides")
+        return globalRecipeMap["overrides-identifiers"][identifier]
+    if identifier in globalRecipeMap["identifiers"]:
+        log(f"Found {identifier} in recipe map")
+        return globalRecipeMap["identifiers"][identifier]
+    return None
+
+
+def find_recipe_by_name_in_map(
+    name: str, skip_overrides: bool = False
+) -> Optional[str]:
+    """Search recipe map for a shortname"""
+    # Check the overrides first, unless skipping them
+    if not skip_overrides and name in globalRecipeMap["overrides"]:
+        log(f"Found {name} in recipe map overrides")
+        return globalRecipeMap["overrides"][name]
+    # search by "Name" in the recipe map
+    if name in globalRecipeMap["shortnames"]:
+        log(f"Found {name} in recipe map")
+        return globalRecipeMap["shortnames"][name]
+    return None
+
+
+def find_name_from_identifier(identifier: str) -> Optional[str]:
+    """Find a recipe name from its identifier"""
+    # TODO: change this to Recipe object
+    recipe_path = globalRecipeMap["identifiers"].get(identifier)
+    for shortname, path in globalRecipeMap["shortnames"].items():
+        if recipe_path == path:
+            return shortname
+    log_err(f"Could not find shortname from {identifier}!")
+    return None
+
+
+def find_identifier_from_name(name: str) -> Optional[str]:
+    """Find a recipe identifier from its shortname"""
+    # TODO: change this to Recipe object
+    recipe_path = globalRecipeMap["shortnames"].get(name)
+    for id, path in globalRecipeMap["identifiers"].items():
+        if recipe_path == path:
+            return id
+    log_err(f"Could not find identifier from {name}!")
+    return None
+
+
+if __name__ == "__main__":
+    read_recipe_map()
+    print("** Building chain for GoogleChromePkg.pkg")
+    # chain = RecipeChain()
+    # chain.add_recipe(
+    #     "/Users/nmcspadden/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes/GoogleChrome/GoogleChromePkg.pkg.recipe"
+    # )
+    # chain.build()
+    # chain.display_chain()
+    # print("** Check-only processors:")
+    # print(chain.get_check_only_processors())
+    # recipe = fetch_recipe_chain("GoogleChromePkg.pkg", check_only=True)
+    # recipe.display_chain()
+    recipe = fetch_recipe_chain("Firefox.munki", check_only=False)
+    recipe.display_chain()
+    # print("** Dictionary version")
+    # rdict = recipe.to_dict()
+    # pprint.pprint(rdict, width=1)
+    # print("** Checking non-git recipe")
+    # non_git_recipe = "/Users/nmcspadden/Library/AutoPkg/Recipes/AutoPkg-Test.download.recipe"
+    # hash = get_git_commit_hash(non_git_recipe)
+    # print(f"Git hash: {hash}")
```

Comment:
Not sure how much of this you want to keep in the final PR.

Reply:
Yeah this will all come out. Was part of the testing process.